airflow/operators/python.py

```diff
@@ -165,7 +166,7 @@ def __init__(
         multiple_outputs: bool = False,
         **kwargs,
     ) -> None:
-        kwargs['task_id'] = self._get_unique_task_id(task_id, kwargs.get('dag'))
+        kwargs['task_id'] = self._get_unique_task_id(task_id, kwargs.get('dag'), kwargs.get('task_group'))
```

Comment:
Instead of doing `kwargs.get`, I suggest having `dag` and `task_group` as two keyword arguments with defaults. I.e. like this:

```python
    def __init__(
        self,	        self,
        *,	        *,
        python_callable: Callable,	        python_callable: Callable,
        task_id: str,	        task_id: str,
        op_args: Tuple[Any],	        op_args: Tuple[Any],
        op_kwargs: Dict[str, Any],	        op_kwargs: Dict[str, Any],
        multiple_outputs: bool = False,	        multiple_outputs: bool = False,
        dag=None,
        task_group=None,
        **kwargs,	        **kwargs,
    ) -> None:	    ) -> None:
        kwargs['task_id'] = self._get_unique_task_id(task_id, dag, task_group)
        ...
        super().__init__(dag=dag, task_group=task_group, **kwargs)
```

Reply:
That was something I wanted suggest when we were doing the `@task` PR. However, I like that we access the kwargs and then we pass them to super constructor. Additionally they are not visible in signature so users don't have to think about them. But no strong opinion on my side